File: model/hall.js
// const Joi = require("joi");
const mongoose = require("mongoose");

const Hall = mongoose.model(
  "Hall",
  new mongoose.Schema({
    name: {
      type: String,
      required: true
    },
    abbreviation: {
      type: String,
      required: true
    },
    imgUrl: {
      type: String,
      required: true
    },
    colourCode: {
      type: String,
      required: true
    },
    malePoint: {
      type: Number,
      required: true,
      default: 0,
      min: 0
    },
    femalePoint: {
      type: Number,
      required: true,
      default: 0,
      min: 0
    },
    totalPoint: {
      type: Number,
      required: true,
      default: 0,
      min: 0
    }
  })
);

function validateHall(hall) {
  const schema = {};

  return Joi.validate(hall, schema);
}

exports.Hall = Hall;
exports.validate = validateHall;

File: pkg/models/hall.go
package models

import "go.mongodb.org/mongo-driver/v2/bson"

type Hall struct {
	ID           bson.ObjectID `bson:"_id,omitempty" json:"_id,omitempty"`
	Name         string        `bson:"name" json:"name"`
	Abbreviation string        `bson:"abbreviation" json:"abbreviation"`
	ImgUrl       string        `bson:"imgUrl" json:"imgUrl"`
	ColourCode   string        `bson:"colourCode" json:"colourCode"`
	MalePoint    float64       `bson:"malePoint" json:"malePoint"`
	FemalePoint  float64       `bson:"femalePoint" json:"femalePoint"`
	TotalPoint   float64       `bson:"totalPoint" json:"totalPoint"`
}

// HallStandingUpdate mirrors the fields the bulk PUT /api/hall endpoint accepts per hall.
type HallStandingUpdate struct {
	ID          bson.ObjectID `bson:"_id" json:"_id"`
	MalePoint   *float64      `bson:"malePoint,omitempty" json:"malePoint,omitempty"`
	FemalePoint *float64      `bson:"femalePoint,omitempty" json:"femalePoint,omitempty"`
	TotalPoint  *float64      `bson:"totalPoint,omitempty" json:"totalPoint,omitempty"`
}
